src/main/java/Engine.java

```diff
@@ -5,52 +5,94 @@ public class Engine {
     private final TaskList taskList;
 
     private enum Command {
-        ERROR,
         EXIT,
-        ECHO,
         LIST,
-        ADD
+        ADD,
+        MARK,
+        UNMARK,
+
+        // errors
+        ERROR
     }
 
     Engine() {
         this.scanner = new Scanner(System.in);
         this.taskList = new TaskList();
     }
 
-    private String getOutput(Command command, String text) {
+    private String getOutput(Command command, String args) {
         switch(command) {
```

Comment:
i like that you use switch case instead of if-else statements since it improves overall code readability